core/workspace_svg.js

```diff
@@ -1482,7 +1482,9 @@ Blockly.WorkspaceSvg.prototype.updateToolbox = function(tree) {
     this.options.languageTree = tree;
     this.flyout_.show(tree.childNodes);
   }
-  this.toolbox_.position();
```

Comment:
move `this.toolbox_.position();` to just after line 1477.

Context: the if statement on line 1472 checks whether this is a toolbox with categories, and if so it populates the toolbox.  Positioning the toolbox is a reasonable followup to that, and means you don't need an extra if.

You may also need to call `this.flyout_.position()` after line 1483, which is the equivalent for the non-category toolbox.